src/core/qgsattributetableconfig.cpp

```diff
@@ -198,6 +198,14 @@ void QgsAttributeTableConfig::readXml( const QDomNode &node )
 
   mSortExpression = configNode.toElement().attribute( QStringLiteral( "sortExpression" ) );
   mSortOrder = static_cast<Qt::SortOrder>( configNode.toElement().attribute( QStringLiteral( "sortOrder" ) ).toInt() );
+  switch ( mSortOrder )
```

Comment:
I'd simply write the following. Same effect, easier to read and shorter :)

    if ( mSortOrder != Qt::AscendingOrder && mSortOrder != Qt::DescendingOrder )